### DISCUSSION TITLE
Pekko alternative of akka-stream-contrib?

### DISCUSSION THREAD

**Discussion by kaplanbar** [apache/pekko#580] [General]
Hello, is there an alternative for akka-stream-contrib which uses Pekko?
https://github.com/akka/akka-stream-contrib/tree/main

**Comment by pjfanning** [comment]
Even akka/akka-stream-contrib is not regularly published in jar format. Its README says this:

```
This repository is not released as a binary artifact and only shared as sources.
```

It is usually close to trivial to rewrite Akka code to work with Pekko. How about trying to copy/paste the akka-stream-contrib code you need into your own project and then rewriting it to use Pekko libs instead?

See https://pekko.apache.org/docs/pekko/current/project/migration-guides.html

**Reply by mdedetrich** [reply]
I think at some point in the future we need to evaluate why something like akka-stream-contrib even exists and how its relevant (or not) for Pekko. akka-stream-contrib is in some kind of purgatory state where it contains a collection of utility methods that are not "good enough" for Akka, and yet the source of the project exists however it doesn't even have binary jars.

The fact that it doesn't release binary jar's gives the impression it shouldn't even be a source repo.

**Reply by pjfanning** [reply]
It is more like a samples repo than a lib repo. I see no harm in this.

**Reply by He-Pin** [reply]
I think just because the operator may not  that generic ？

**Comment by pjfanning** [comment]
I created https://github.com/pjfanning/pekko-stream-contrib - I'm not planning on publishing a jar but the source should be usable.

**Comment by He-Pin** [comment]
@kaplanbar What specified operators do you need? I think unlink zio or reactor/rxjava, pekko lack some operators and not easy to use.
